src/Template/Layout/default.ctp

```diff
@@ -70,7 +74,7 @@ $htmlDir = LanguagesLib::getLanguageDirection($lang);
     <link rel="search" type="application/opensearchdescription+xml"
           href="/opensearch.xml" title="Tatoeba" />
     
-    <?php if ($controller === 'pages' && $action === 'index' && !CurrentUser::isMember()) { ?>
+    <?php if ($isHomepage && !CurrentUser::isMember() || $isSearch) { ?>
         <meta name="viewport" content="width=device-width, initial-scale=1.0">     
```

Comment:
Since we are going to need that viewport meta code on more and more pages, I think you could make good use of [extended views](https://book.cakephp.org/3/en/views.html#extending-views) here. I’ve never used that myself, but it should allow `layout.ctp` to display variables that are defined by child views.

Reply:
Actually there won't be the need to use extended views here. I just realized we can already use in the layout variables set in the views. We did this with `title_for_layout`.

But thanks for mentioning this mechanism from CakePHP! That can help a lot for future redesign 😄